app/models/camaleon_cms/custom_field_group.rb

```diff
@@ -69,52 +81,55 @@ def add_fields(items, item_options)
         when 'PostType_Category'
-          caption = "Fields for Categories in <b>#{self.site.post_types.find(self.objectid).decorate.the_title}</b>"
+          caption = "Fields for Categories in <b>#{site.post_types.find(objectid).decorate.the_title}</b>"
         when 'PostType_PostTag'
-          caption = "Fields for Post tags in <b>#{self.site.post_types.find(self.objectid).decorate.the_title}</b>"
+          caption = "Fields for Post tags in <b>#{site.post_types.find(objectid).decorate.the_title}</b>"
         when 'Widget::Main'
-          caption = "Fields for Widget <b>(#{CamaleonCms::Widget::Main.find(self.objectid).name.translate})</b>"
+          caption = "Fields for Widget <b>(#{CamaleonCms::Widget::Main.find(objectid).name.translate})</b>"
         when 'Theme'
-          caption = "Field settings for Theme <b>(#{self.site.themes.find(self.objectid).name rescue self.objectid})</b>"
+          caption = "Field settings for Theme <b>(#{site.themes.find(objectid).name rescue objectid})</b>"
         when 'NavMenu'
-          caption = "Field settings for Menus <b>(#{CamaleonCms::NavMenu.find(self.objectid).name})</b>"
+          caption = "Field settings for Menus <b>(#{CamaleonCms::NavMenu.find(objectid).name})</b>"
         when 'Site'
           caption = "Field settings the site"
         when 'PostType'
           caption = "Fields for all <b>Post_Types</b>"
         when 'Post'
-          p = CamaleonCms::Post.find(self.objectid).decorate
+          p = CamaleonCms::Post.find(objectid).decorate
           caption = "Fields for content <b>(#{p.the_title})</b>"
         else # 'Plugin' or other class
-          caption = "Fields for <b>#{self.object_class}</b>"
+          caption = "Fields for <b>#{object_class}</b>"
       end
     rescue => e
-      Rails.logger.info "----------#{e.message}----#{self.attributes}"
+      Rails.logger.info "----------#{e.message}----#{attributes}"
     end
     caption
   end
 
   private
+
   def before_validating
-    self.slug = "_group-#{self.name.to_s.parameterize}" unless self.slug.present?
+    slug = "_group-#{name.to_s.parameterize}" unless slug.present?
   end
 
   # auto save the default field values
   def auto_save_default_values(field, options)
     options = options.with_indifferent_access
-    class_name = self.object_class.split("_").first
-    if ["Post", "Category", "Plugin", "Theme"].include?(class_name) && self.objectid.present? && (options[:default_value].present? || options[:default_values].present?)
+    class_name = object_class.split("_").first
+    if %w(Post Category Plugin Theme).include?(class_name) && objectid.present? &&
+      (options[:default_value].present? || options[:default_values].present?)
       if class_name == "Theme"
-        owner = "CamaleonCms::#{class_name}".constantize.where(id: self.objectid).first # owner model
+        owner = "CamaleonCms::#{class_name}".constantize.find(objectid) # owner model
```

Comment:
`find()` stands for find by id, so `where(id: ).first` can also be written as `find()`. To avoid `ActiveRecord::RecordNotFound ` it can be change into finy_by(id:), which return `nil` if no records found